test/e2e/annotations/cors.go

```diff
@@ -96,6 +97,42 @@ var _ = framework.DescribeAnnotation("cors-*", func() {
 			})
 	})
 
+	ginkgo.It("should preserve upstream vary headers", func() {
+		f.NewHttpbunDeployment(framework.WithDeploymentName("cors-service"))
+
+		host := corsHost
+		origin := "https://origin.cors.com"
```

Comment:
just a thought, shouldn't the vary header only be necessary if the origin is dynamic?
i.e. a wildcard?

https://*.origin.cors.com

Reply:
It is needed when the origin is dynamic, or when a list is specified. However, the current implementation adds it in every CORS response as it's a bit complex to determine this set of conditions. WDYT?

Reply:
The only thing is, if this header is automatically added to every request, then other ones unit tests will most likely fail.

It might be me that is unsure, but in your test I would've expected to see a wildcard or a list (like you mentioned :)). I see the single domain like above as static

Reply:
I updated the test using a subdomain wildcard. It looks like other tests are passing, which one were you talking about?